Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -904,7 +905,9 @@ open class NavigationMapView: UIView {
                     try mapView.mapboxMap.style.addSource(arrowSource, id: NavigationMapView.SourceIdentifier.arrowSource)
                     arrowLayer.source = NavigationMapView.SourceIdentifier.arrowSource
                     
-                    if mapView.mapboxMap.style.sourceExists(withId: NavigationMapView.LayerIdentifier.waypointCircleLayer) {
+                    if allLayerIds.contains(NavigationMapView.LayerIdentifier.puck3DLayer) {
```

Comment:
If layer already exists why're we adding it, instead of updating?

Reply:
It is to check whether the 3D puck layer exists, if the 3D puck layer exists, we add the `arrowLayer` below the puck3D layer. It's the same case for `arrowSymbolLayer`, to add it below the 3D puck layer.

Reply:
The CarPlay issue with maneuvers should be related with the layerID problem as [below](https://github.com/mapbox/mapbox-navigation-ios/pull/3106#issuecomment-867119146) , add the check of layer existence before set layer position. Similar issue happened for the way point circle layer.